frontend/src/utils/documentReferencesHelper.js

```diff
@@ -0,0 +1,90 @@
+export function mergePartialGroupToGroupsArray(groups, partialGroup) {
+  if (!partialGroup) {
+    return groups;
+  }
+
+  let result = [];
+  let partialGroupMerged = false;
+
+  for (const group of groups) {
```

Comment:
Maybe a simpler solution could be used either using reduce https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce. or  with some lodash utils. Looks big. Not mandatory to do that